scripts/csv_importer.py

```diff
@@ -0,0 +1,273 @@
+    name_field = headers[0]
+    print(f"\nHeaders: {list(headers)}")
+    print(f"Default item name will use the value in column: {name_field}.")
+    if input("Use a different item name? (y/n) ") == 'y':
+        name_field = None
+        while name_field not in headers:
+            name_field = input("Item name: ")
+
+    print("By default all fields will be public.")
+    print("Enter names of fields that should be made private, separated by commas (only commas, not comma+space!).")
+    private_fields = input(">>").split(',')
+
+    cursor.execute("SELECT title FROM koi_collection")
+    collection_names = [name[0] for name in cursor.fetchall()]
+
+    if collection in collection_names:
+        create_new_collection = False
+    else:
+        create_new_collection = True
+
+    collection_name = collection
+
+    cursor.execute("SELECT username FROM koi_user")
+    usernames = [name[0] for name in cursor.fetchall()]
+    if len(usernames) == 1:
+        username = usernames[0]
+    else:
+        print("\nUsers:")
+        for name in usernames:
+            print(name)
+
+        username = None
+        while username not in usernames:
+            username = input("Collection owner name: ")
+
+    cursor.execute(f"SELECT id FROM koi_user WHERE username='{username}'")
+    owner_id = cursor.fetchone()[0]
+
+    if create_new_collection:
+        collection_id = insert_collection(owner_id, collection_name)
+        insert_log(owner_id, collection_id, collection_name, "App\\Entity\\Collection")
+    else:
+        while collection_name not in collection_names:
+            collection_name = input("Collection name: ")
+            print(collection_names, type(collection_names))
+
+        cursor.execute(f"SELECT id FROM koi_collection WHERE title='{collection_name}'")
+        collection_id = cursor.fetchone()[0]
+
+    for item in items:
+        item_id = insert_item(owner_id, collection_id, item[name_field])
+
+        for index, attribute_name in enumerate(headers):
+            if attribute_name != name_field:
+                insert_datum(owner_id, item_id, attribute_name, item[attribute_name], index + 1,
+                             "private" if attribute_name in private_fields else "public")
+
+        insert_log(owner_id, item_id, item[name_field], "App\\Entity\\Item")
+
+    cursor.execute(f"SELECT cached_values FROM koi_collection WHERE id='{collection_id}'")
+    cached_values = ast.literal_eval(cursor.fetchone()[0])
+    cached_values["counters"]["items"] += len(items)
+    cursor.execute(f"UPDATE koi_collection SET cached_values=%s, updated_at=%s WHERE id='{collection_id}'",
+                   (json.dumps(cached_values), get_current_time()))
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="Import data from a csv file to Koillection database")
+    parser.add_argument("-u", "--user", nargs='?', help="Database user", type=str)
```

Comment:
This should contain username from compose, which is root/password
`default=root`
Please follow below.